### PR TITLE
FIRECERT-2044-New Default validation for Firebolt responds step

### PR DESCRIPTION
**Pull request by anjalimukundan**

### CODE REVIEW

File: cypress/fixtures/fireboltCalls/accessibility.json [hunk 1]
```diff
@@ -10,215 +10,321 @@
     },
     "DISABLE_CLOSEDCAPTIONS":{
         "method": "manage_closedcaptions.setEnabled",
-		"params": "FALSE"
+        "params": "FALSE",
+        "validationJsonPath": "result",
+        "expected": "NULL"
     },
```
Comment (neeradanelxsi) [inline cypress/fixtures/fireboltCalls/accessibility.json:16]: why are we adding the **validationJsonPath** and **expected** values here ? The purpose of this ticket is to reuse the same key  for setter call and setter validation. And the code should have the capability to handle if no **validationJsonPath** and **expected** are given here

File: cypress/TestCases/FireboltCertification/Accessibility.feature [hunk 2]
```diff
@@ -9,54 +9,54 @@ Feature: Accessibility
...
    @Accessibility @coreSDK @sdk @transport
    Scenario Outline: Accessibility.closedCaptionsSettings - Positive Scenario: <Scenario> with 'null' params
       When '3rd party app' registers for the 'accessibility onClosedCaptionsSettingsChanged' event using the 'Firebolt' API
       And '3rd party app' invokes the 'Firebolt' API to 'get closedCaptions settings'
       And 1st party app invokes the 'Firebolt' API to '<Key>'
-      Then 'Firebolt' platform responds to '1st party app' with '<Set_Method_Content>'
+      Then 'Firebolt' platform responds to '1st party app' for '<Set_Method_Content>'
       When '3rd party app' invokes the 'Firebolt' API to 'get closedCaptions settings'
-      Then 'Firebolt' platform responds with '<Method_Content>'
+      Then 'Firebolt' platform responds for '<Method_Content>'
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Accessibility.feature:59]: Line 57 has to be changed to to use the same key as the setter call which is in line 56

File: cypress/TestCases/FireboltCertification/Advertising.feature [hunk 3]
```diff
@@ -39,23 +39,23 @@ Feature: Advertising
...
       And '3rd party app' invokes the 'Firebolt' API to 'get initialization parameters'
       When 1st party app invokes the 'Firebolt' API to '<API_Key>'
-      Then 'Firebolt' platform responds to '1st party app' with 'null for privacy setAllowAppContentAdTargeting'
+      Then 'Firebolt' platform responds to '1st party app' for '<API_Key>'
       When '3rd party app' invokes the 'Firebolt' API to 'get advertisingId'
       And '3rd party app' invokes the 'Firebolt' API to 'get initialization parameters'
-      Then 'Firebolt' platform responds with '<AdvertidingId_Validation_key>'
-      And 'Firebolt' platform responds with '<Parameters_Validation_key>'
+      Then 'Firebolt' platform responds for '<AdvertidingId_Validation_key>'
+      And 'Firebolt' platform responds for '<Parameters_Validation_key>'
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Advertising.feature:58]: No need to change the glue to use **for** instead of **with** in case of getter

File: cypress/support/step_definitions/validations.js [hunk 4]
```diff
@@ -49,10 +49,13 @@ Given(
         fireboltItems.forEach((item) => {
           const validationType = item.event ? CONSTANTS.EVENT : CONSTANTS.METHOD;
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:51]: we cannot keep manage_ in the code. it would be proprietary. That has to be removed

File: cypress/support/step_definitions/validations.js [hunk 5]
```diff
@@ -49,10 +49,13 @@ Given(
...
+          let methodOrEvent = item[validationType];
+          methodOrEvent = methodOrEvent.includes('manage_')
+            ? methodOrEvent.replace('manage_', '')
+            : methodOrEvent;
           const context = item.context ? item.context : CONSTANTS.NO_CONTEXT;
-          const validationJsonPath = item.validationJsonPath;
-          const expected = item.expected;
+          const validationJsonPath = item.validationJsonPath ? item.validationJsonPath : 'result';
+          const expected = item.expected ? item.expected : 'NULL';
           const expectingError = item.expectingError;
```
[the remark this answers is not in the document]
Comment (anjalimukundan) [reply cypress/support/step_definitions/validations.js:59]: moved to constants

File: cypress/support/step_definitions/validations.js [hunk 6]
```diff
@@ -29,16 +29,16 @@ import UTILS from '../cypress-support/src/utils';
...
  * Given 'Firebolt' platform triggers event 'Validate device id'
  * Given 'Firebolt' platform triggers to '1st party app' event 'Validate device id'
  * Given 'Firebolt' platform triggers to 'test.test.test' event 'Validate device id'
  * Given 'Firebolt' platform does not trigger event for 'onclosedCaptionsSettingsChanged'
  */
 
 Given(
-  /'(.+)' platform (responds|triggers|does not trigger)(?: to '(.+)')? (with|event)(?: for)? '(.+)'$/,
+  /'(.+)' platform (responds|triggers|does not trigger)(?: to '(.+)')? (for|event)(?: for)? '(.+)'$/,
   async (sdk, eventExpected, appId, event, key) => {
```
[the remark this answers is not in the document]
Comment (anjalimukundan) [reply cypress/support/step_definitions/validations.js:42]: updated

File: cypress/fixtures/fireboltCalls/accessibility.json (hunk 1 above)
Comment (anjalimukundan) [reply cypress/fixtures/fireboltCalls/accessibility.json:16]: reverted the changes

File: cypress/support/step_definitions/validations.js (hunk 4 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/validations.js:51]: updated the code

File: cypress/TestCases/FireboltCertification/Advertising.feature (hunk 3 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Advertising.feature]: updated

File: cypress/TestCases/FireboltCertification/Accessibility.feature (hunk 2 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Accessibility.feature]: updated

File: cypress/TestCases/FireboltCertification/Advertising.feature [hunk 7]
```diff
@@ -9,7 +9,7 @@ Feature: Advertising
       When '3rd party app' registers for the 'advertising onPolicyChanged' event using the 'Firebolt' API
       And '3rd party app' invokes the 'Firebolt' API to 'get advertising policy'
       And 1st party app invokes the 'Firebolt' API to '<API_Key>'
-      Then 'Firebolt' platform responds to '1st party app' with 'null for privacy setAllowAppContentAdTargeting'
+      Then 'Firebolt' platform responds to '1st party app' with '<API_Key>'
       When '3rd party app' invokes the 'Firebolt' API to 'get advertising policy'
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Advertising.feature:13]: **_with_** need to be replaced with **_for_**

File: cypress/TestCases/FireboltCertification/DiscoveryLaunchCold.feature [hunk 8]
```diff
@@ -115,4 +115,4 @@ Feature: Discovery.launch_ColdLaunch
         Examples:
             | Scenario              | Discovery_Launch_Key                   | Call_Parameters_Initialization_With_Context_Key   | Validation_Key_For_Parameters_Initialization_With_Intent |
             | Passing random string | launch app with null intent source     | get initialization parameters for null intent     | nullintent source for initialization parameters          |
-            | Passing valid string  | launch app with playback intent source | get initialization parameters for playback intent | playbackintent source for initialization parameters      |
+            | Passing valid string  | launch app with playback intent source | get initialization parameters for playback intent | playbackintent source for initialization parameters      |
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/DiscoveryLaunchCold.feature:118]: There are many feature files which doesn't have any changes related to this ticket. Is it possible to remove those line additions ?  (Unless its a formatting change that was added automatically)

cypress/TestCases/FireboltCertification/DiscoveryLaunchCold.feature
cypress/TestCases/FireboltCertification/DiscoveryLaunchHot.feature
cypress/TestCases/FireboltCertification/Keyboard.feature 
cypress/TestCases/FireboltCertification/LifecycleTestcases/LifecycleBackground.feature

File: cypress/TestCases/FireboltCertification/Manage/AccountManage.feature [hunk 9]
```diff
@@ -7,18 +7,18 @@ Feature: Account_Manage
...
       When 1st party app invokes the 'Firebolt' API to 'push session token'
       And '3rd party app' invokes the 'Firebolt' API to 'get advertisingId with custom error'
-      Then 'Firebolt' platform responds to '1st party app' with 'null for account session'
+      Then 'Firebolt' platform responds to '1st party app' for 'push session token'
 
    @Account @manageSDK @sdk @transport
    Scenario Outline: Account.session - Negative Scenario: <Scenario> expecting error
       When 1st party app invokes the 'Firebolt' API to '<Key>'
-      Then 'Firebolt' platform responds to '1st party app' with 'invalid parameters for account session'
+      Then 'Firebolt' platform responds to '1st party app' for 'invalid parameters for account session'
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Manage/AccountManage.feature:21]: Revert the changes in line 21.

File: cypress/TestCases/FireboltCertification/Manage/AdvertisingManage.feature [hunk 10]
```diff
@@ -7,14 +7,14 @@ Feature: Advertising_Manage
    @Advertising @manageSDK @sdk @transport
    Scenario: Advertising.resetIdentifier - Positive Scenario: Reset Identifier
       When 1st party app invokes the 'Firebolt' API to 'reset identifier for advertising'
-      Then 'Firebolt' platform responds to '1st party app' with 'null for advertising resetIdentifier'
+      Then 'Firebolt' platform responds to '1st party app' with 'reset identifier for advertising'
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Manage/AdvertisingManage.feature:10]: revert the changes in line 10

File: cypress/TestCases/FireboltCertification/Manage/AdvertisingManage.feature [hunk 11]
```diff
@@ -30,12 +30,12 @@ Feature: Advertising_Manage
    @Advertising @manageSDK @sdk @transport
    Scenario: Advertising.setSkipRestriction - Negative Scenario: SkipRestriction expecting error
       When 1st party app invokes the 'Firebolt' API to 'set skipRestriction with integer'
-      Then 'Firebolt' platform responds to '1st party app' with 'invalid parameters for skipRestriction'
+      Then 'Firebolt' platform responds to '1st party app' for 'invalid parameters for skipRestriction'
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Manage/AdvertisingManage.feature:33]: revert the changes in line 33

File: cypress/TestCases/FireboltCertification/Manage/AudioDescriptionsManage.feature [hunk 12]
```diff
@@ -22,7 +22,7 @@ Feature: AudioDescriptions_Manage
    @AudioDescriptions @manageSDK @sdk @transport
    Scenario Outline: AudioDescriptions.enabled - Negative Scenario: <Scenario> expecting error
       When 1st party app invokes the 'Firebolt' API to '<API_Key>'
-      Then 'Firebolt' platform responds to '1st party app' with 'invalid params for audiodescriptions setEnabled'
+      Then 'Firebolt' platform responds to '1st party app' for 'invalid params for audiodescriptions setEnabled'
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Manage/AudioDescriptionsManage.feature:25]: revert the changes in line 25

File: cypress/TestCases/FireboltCertification/Manage/ClosedCaptionsManage.feature [hunk 13]
```diff
@@ -74,30 +74,30 @@ Feature: ClosedCaptions_Manage
...
+            | textAlignVertical | textAlignVertical | set textAlignVertical as null | get textAlignVertical | middle for textAlignVertical in closedCaptions    | onTextAlignVerticalChanged for closedcaptions with middle    | closedcaptions onTextAlignVerticalChanged |
+            | windowColor       | windowColor       | set windowColor as null       | get windowColor       | #000000 for windowColor in closedCaptions         | onWindowColorChanged for closedcaptions with #000000         | closedcaptions onWindowColorChanged       |
+            | windowOpacity     | windowOpacity     | set windowOpacity as null     | get windowOpacity     | 0 for windowOpacity in closedCaptions             | onWindowOpacityChanged for closedcaptions with 0             | closedcaptions onWindowOpacityChanged     |
 
     @ClosedCaptions @manageSDK
     Scenario Outline: ClosedCaptions.<Method> - Negative Scenario: <Scenario> expecting error
         When 1st party app invokes the 'Firebolt' API to '<API_Key>'
-        Then 'Firebolt' platform responds to '1st party app' with '<Method_Validation_key>'
+        Then 'Firebolt' platform responds to '1st party app' for '<Method_Validation_key>'
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Manage/ClosedCaptionsManage.feature:101]: revert the changes in 100

File: cypress/TestCases/FireboltCertification/DiscoveryLaunchCold.feature (hunk 8 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/DiscoveryLaunchCold.feature:118]: it is added automatically. Tried removing them.. bt not fixing

File: cypress/TestCases/FireboltCertification/Capabilities.feature [hunk 14]
```diff
@@ -226,7 +225,7 @@ Feature: Capabilities
     And User set response for 'set keyboard response in platform'
     And Framework registers 'keyboard' test provider
     When 1st party app invokes the 'Firebolt' API to 'check availability of keyboard capability'
-    Then 'Firebolt' platform responds to '1st party app' with 'true for keyboard capability availability'
+    Then 'Firebolt' platform responds to '1st party app' for 'check availability of keyboard capability'
     And 'Firebolt' platform triggers to '1st party app' event 'expected value for capabilities onAvailable'
```
[the remark this answers is not in the document]
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Capabilities.feature:229]: reverted capabilities changes

File: cypress/TestCases/FireboltCertification/Manage/AccountManage.feature (hunk 9 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Manage/AccountManage.feature:21]: reverted for negative scenarios

File: cypress/TestCases/FireboltCertification/Manage/AdvertisingManage.feature (hunk 10 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Manage/AdvertisingManage.feature:10]: added the previous value

File: cypress/TestCases/FireboltCertification/Manage/AdvertisingManage.feature (hunk 11 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Manage/AdvertisingManage.feature:33]: reverted to "with"

File: cypress/TestCases/FireboltCertification/Manage/ClosedCaptionsManage.feature (hunk 13 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Manage/ClosedCaptionsManage.feature:101]: reverted to "with"

File: cypress/TestCases/FireboltCertification/Advertising.feature [hunk 15]
```diff
@@ -24,7 +24,7 @@ Feature: Advertising
       When '3rd party app' registers for the 'advertising onPolicyChanged' event using the 'Firebolt' API
       And '3rd party app' invokes the 'Firebolt' API to 'get advertising policy'
       And 1st party app invokes the 'Firebolt' API to '<API_Key>'
-      Then 'Firebolt' platform responds to '1st party app' with 'null for advertising setSkipRestriction'
+      Then 'Firebolt' platform responds to '1st party app' with '<API_Key>'
```
Comment (kschrief) [inline cypress/TestCases/FireboltCertification/Advertising.feature:27]: Are these supposed to be "responds FOR" for a default response?

File: cypress/support/step_definitions/validations.js [hunk 16]
```diff
@@ -49,10 +49,13 @@ Given(
         fireboltItems.forEach((item) => {
           const validationType = item.event ? CONSTANTS.EVENT : CONSTANTS.METHOD;
 
-          const methodOrEvent = item[validationType];
+          let methodOrEvent = item[validationType];
+          methodOrEvent = methodOrEvent.includes('_') ? methodOrEvent.split('_')[1] : methodOrEvent;
           const context = item.context ? item.context : CONSTANTS.NO_CONTEXT;
-          const validationJsonPath = item.validationJsonPath;
-          const expected = item.expected;
+          const validationJsonPath = item.validationJsonPath
```
Comment (kschrief) [inline cypress/support/step_definitions/validations.js:55]: Can we put a quick comment here to explain why we're setting "result" and "null" as defaults?

**Review by kschrief — CHANGES_REQUESTED** [review]

File: cypress/TestCases/FireboltCertification/Advertising.feature (hunk 15 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Advertising.feature:27]: updated as :  responds to '1st party app' for '<API_Key>'

File: cypress/support/step_definitions/validations.js (hunk 16 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/validations.js:55]: added comment as :
          // if the validationObject doesn't have validationJsonPath or expected fields, assign default values.

**Review by neeradanelxsi — DISMISSED** [review]

**Review by kschrief — DISMISSED** [review]

**Review by Abhishk123 — APPROVED** [review]
